pubsub-mapped-api/src/main/java/com/google/pubsub/common/StubCreator.java

```diff
@@ -0,0 +1,38 @@
+package com.google.pubsub.common;
+
+import com.google.pubsub.v1.PublisherGrpc;
+import com.google.pubsub.v1.PublisherGrpc.PublisherBlockingStub;
+import com.google.pubsub.v1.SubscriberGrpc;
+import com.google.pubsub.v1.SubscriberGrpc.SubscriberBlockingStub;
+import com.google.pubsub.v1.SubscriberGrpc.SubscriberFutureStub;
+import java.util.concurrent.TimeUnit;
+
+public class StubCreator {
+
+  private ChannelUtil channelUtil;
+
+  public StubCreator(ChannelUtil channelUtil) {
+    this.channelUtil = channelUtil;
+  }
+
+  public SubscriberBlockingStub getSubscriberBlockingStub() {
+    return SubscriberGrpc.newBlockingStub(channelUtil.getChannel())
+        .withCallCredentials(channelUtil.getCallCredentials());
+  }
+
+  public SubscriberFutureStub getSubscriberFutureStub() {
+    return SubscriberGrpc.newFutureStub(channelUtil.getChannel())
+        .withCallCredentials(channelUtil.getCallCredentials());
+  }
+
+  public SubscriberFutureStub getSubscriberFutureStub(long timeoutMilliseconds) {
+    return SubscriberGrpc.newFutureStub(channelUtil.getChannel())
+        .withCallCredentials(channelUtil.getCallCredentials())
+        .withDeadlineAfter(timeoutMilliseconds, TimeUnit.MILLISECONDS);
+  }
+
+  public PublisherBlockingStub getPublisherBlockingStub() {
+    return PublisherGrpc.newBlockingStub(channelUtil.getChannel())
```

Comment:
Stubs are expensive to create, we should not be creating them on every call. Other implementations use a list of them that it operates on in round robin. Since this is Java I recommend using the client library found at github.com/GoogleCloudPlatform/google-cloud-java instead of reimplementing Pub/Sub in gRPC.